cmd/run.go

```diff
@@ -572,3 +567,62 @@ func logVersion() {
 		"os/arch", fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
 	)
 }
+
+func forkIDIntervals(ctx context.Context, st *state.State, etherman *etherman.Client, genesisBlockNumber uint64) ([]state.ForkIDInterval, error) {
+	dbTx, err := st.BeginStateTransaction(ctx)
+	if err != nil {
+		return []state.ForkIDInterval{}, fmt.Errorf("error creating dbTx. Error: %v", err)
+	}
+	numberForkIDs, err := st.CountForkIDs(ctx, dbTx)
+	if err != nil {
+		return []state.ForkIDInterval{}, fmt.Errorf("error checking forkID table. Error: %v", err)
+	}
+	log.Debug("numberForkIDs: ", numberForkIDs)
+	var forkIDIntervals []state.ForkIDInterval
+	if numberForkIDs == 0 {
+		// Get last L1block Synced
+		lastBlock, err := st.GetLastBlock(ctx, dbTx)
+		if err != nil && !errors.Is(err, state.ErrStateNotSynchronized) {
+			return []state.ForkIDInterval{}, fmt.Errorf("error checking lastL1BlockSynced. Error: %v", err)
+		}
+		if lastBlock != nil {
```

Comment:
I think we can change this if/else block code for this one, since the if and else code are the same, since I understand also that we need to store the forkid interval in the case of the initial fork (genesis):

```
lastBlockNumber := genesisBlockNumber
if lastBlock != nil {
   lastBlockNumber := lastBlock.BlockNumber
}

log.Info("Getting forkIDs intervals. Please wait...")
// Read Fork ID FROM POE SC
forkIntervals, err := etherman.GetForks(ctx, genesisBlockNumber, lastBlockNumber)
if err != nil {
   return []state.ForkIDInterval{}, fmt.Errorf("error getting forks. Please check the configuration. Error: %v", err)
} else if len(forkIntervals) == 0 {
   return []state.ForkIDInterval{}, fmt.Errorf("error: no forkID received. It should receive at least one, please check the configuration...")
}
log.Info("Storing forkID intervals into db")
// Store forkIDs
dbTx, err := st.BeginStateTransaction(ctx)
if err != nil {
   return []state.ForkIDInterval{}, fmt.Errorf("error creating dbTx. Error: %v", err)
}
for _, f := range forkIntervals {
   err := st.AddForkID(ctx, f, dbTx)
   if err != nil {
   err = dbTx.Rollback(ctx)
   if err != nil {
      return []state.ForkIDInterval{}, fmt.Errorf("error when rollback dbTx. Error: %v", err)
   }
      return []state.ForkIDInterval{}, fmt.Errorf("error adding forkID to db. Error: %v", err)
   }
}
err = dbTx.Commit(ctx)
if err != nil {
   return []state.ForkIDInterval{}, fmt.Errorf("error committing dbTx. Error: %v", err)
}
forkIDIntervals = forkIntervals

```